site/content/en/docs/Overview/architecture.md

```diff
@@ -0,0 +1,82 @@
+---
+title: "Architecture"
+linkTitle: "Architecture"
+weight: 1
+description: >
+  Prow is made up of a collection of microservices (aka "Prow components") that work together in a **service cluster**, leveraging one or more **build clusters** to schedule Prow Jobs (or just "jobs").
+---
+
+## Prow in a Nutshell
+
+Prow creates jobs based on various types of events, such as:
+
+- GitHub events (e.g., a new PR is created, or is merged, or a person comments
+"/retest" on a PR),
+
+- Pub/Sub messages,
+
+- time (these are created by **Horologium** and are called **periodic** jobs),
+and
+
+- retesting (triggered by **Tide**).
+
+Jobs are created inside the Service Cluster as Kubernetes Custom Resources.  The
+**Prow Controller Manager** takes triggered jobs and schedules them into a build
+cluster, where they run as Kubernetes pods. **Crier** then reports the results
+back to GitHub.
+
+```mermaid
+flowchart TD
+
+    classDef yellow fill:#ff0
+    classDef cyan fill:#0ff
+    classDef pink fill:#f99
+
+    subgraph Service Cluster["<span style='font-size: 40px;'><b>Service Cluster</b></span>"]
+        Deck:::cyan
+        Prowjobs:::yellow
+        Crier:::cyan
+        Tide:::cyan
+        Horologium:::cyan
+        Sinker:::cyan
+        PCM[Prow Controller Manager]:::cyan
+        Hook:::cyan
+            subgraph Hook
+            PluginCat(["'cat' plugin"])
+            PluginTrigger(["'trigger' plugin"])
+            end
+    end
+
+    subgraph Build Cluster[<b>Build Cluster</b>]
+        Pods[(Pods)]:::yellow
+    end
+
+    style Legend fill:#fff,stroke:#000,stroke-width:4px
+    subgraph Legend["<span style='font-size: 20px;'><b>LEGEND</b></span>"]
+        direction LR
+        k8sResource[Kubernetes Resource]:::yellow
+        prowComponent[Prow Component]:::cyan
+        hookPlugin([Hook Plugin])
+        Other
+    end
+
+    Prowjobs <-.-> Deck <-----> |Serve| prow.k8s.io
+    GitHub ==> |Webhooks| Hook
+    Prowjobs <-.-> Tide --> |Retest and Merge| GitHub
+    Horologium ---> |Create| Prowjobs
+    GitHub ---> |/meow| PluginCat
+    GitHub --> |/retest| PluginTrigger
```

Comment:
These comments are not received separately from webhooks as currently shown and they are routed to all plugins, not just the ones that react.
To indicate these are webhooks I think it would be clearer to have the lines start at the `Webhooks` box rather than `GitHub`. A dotted line or different color might be better than a solid line too since these are examples?

Reply:
To clarify, you'd prefer the lines to look more like this?

```
    GitHub --> Webhook ---> |/meow| PluginCat
    GitHub --> Webhook --> |/retest| PluginTrigger
```

Aside: currently in the diagram the `Webhooks` is not a node (box) but rather just a labeled edge, so that would need to change as well.

BTW this architecture diagram is mostly copied from one of the existing architecture diagrams I found internally.

> > On second thought, doesn't Hook work with Gerrit? People can comment `/retest` and jobs will get triggered there, right?
> 
> Hook only handles GitHub webhooks, Gerrit does not send webhooks. The Gerrit adapter scrapes Gerrit instances and is responsible for triggering jobs based on new commits or job triggering comments like `/retest`. Some code is shared for the comment matching and determining which job to run, but the components are otherwise quite different.
> 
> I have one remaining bit of feedback, but it isn't critical. /hold

Makes sense

Reply:
> To clarify, you'd prefer the lines to look more like this?

That seems closer to reality to me. I mainly just don't like that the diagram currently shows `/meow` and `/retest` as completely separate from (and parallel to) webhooks.

Reply:
Done